### PR TITLE
Untangle

### LINKED ISSUE #266: Decouple from Laravel
It seems this package isn't actually that coupled to laravel. 

Potentially worth decoupling it so it can be used outside laravel :)

### PR DESCRIPTION
**Pull request by reliq**
The changes herein allow us to use this package easily outside of the Laravel ecosystem by way of PHP DI. 

Key points:
- introduces contracts to easily manage dependencies
- includes integration tests with Laravel DI and PHP-DI
- general structure of package is unchanged

Closes #266

### CODE REVIEW

File: src/Twitter.php [hunk 1]
```diff
@@ -39,20 +39,34 @@
...
-        HelpTrait,
-        ListTrait,
-        MediaTrait,
-        SearchTrait,
-        StatusTrait,
-        TrendTrait,
-        UserTrait,
+    use FormattingHelpers;
+    use
+        AccountTrait;
```
Comment (atymic) [inline src/Twitter.php:44]: i think the `use` should be on the same line as the classname?

Comment (reliq) [reply src/Twitter.php:44]: It should indeed. Will check why this formatting is happening and fix it

Comment (reliq) [reply src/Twitter.php]: Fixed

File: src/Twitter.php [hunk 2]
```diff
@@ -206,9 +185,8 @@ public function get(string $endpoint, $parameters = [], $multipart = false, $ext
     }
 
     /**
-     * @param string $endpoint
-     * @param array  $parameters
-     * @param bool   $multipart
+     * @param array $parameters
+     * @param bool  $multipart
```
Comment (atymic) [inline src/Twitter.php:189]: worth typehinting these?

**Comment by atymic** [comment]
Awesome job :D

**Pull request merged**